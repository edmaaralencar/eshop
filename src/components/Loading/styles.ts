import styled from 'styled-components'

export const Wrapper = styled.div`
  @keyframes spinner {
    to {
      transform: rotate(360deg);
    }
  }

  .loading {
    width: 10rem;
    height: 10rem;
    border: 5px solid ${({ theme }) => theme.colors.primary.light};
    border-radius: 50%;
    border-top-color: ${({ theme }) => theme.colors.primary.main};
    animation: spinner 2s linear infinite;
    margin: 0 auto;
  }
`
