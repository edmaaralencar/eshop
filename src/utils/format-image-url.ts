export function formatImageUrl(image: string) {
  return `/products/${image}`
}
