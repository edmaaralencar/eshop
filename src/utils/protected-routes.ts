import { GetServerSidePropsContext } from 'next'
// import { unstable_getServerSession } from 'next-auth'
import { getSession } from 'next-auth/react'
// import { options } from 'pages/api/auth/[...nextauth].api'

async function protectedRoutes(context: GetServerSidePropsContext) {
  // const session = await unstable_getServerSession(
  //   context.req,
  //   context.res,
  //   options
  // )

  // if (!session) {
  //   return {
  //     redirect: {
  //       destination: `/sign-in?callbackUrl=${context.resolvedUrl}`,
  //       permanent: false
  //     }
  //   }
  // }

  const session = await getSession(context)

  if (!session) {
    context.res.setHeader(
      'Location',
      `/sign-in?callbackUrl=${context.resolvedUrl}`
    )
    context.res.statusCode = 302
  }

  return session
}

export default protectedRoutes
