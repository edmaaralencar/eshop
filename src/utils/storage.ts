const APP_KEY = '@eshop'

export function getStorageItem(key: string) {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(`${APP_KEY}.${key}`)
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return JSON.parse(data!)
}

export function setStorageItem(key: string, value: any[]) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(`${APP_KEY}.${key}`, data)
}

export function removeStorageItem(key: string) {
  if (typeof window === 'undefined') return

  return window.localStorage.removeItem(`${APP_KEY}.${key}`)
}
