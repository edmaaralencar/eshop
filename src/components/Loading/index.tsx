import * as S from './styles'

const Loading = () => {
  return (
    <S.Wrapper data-testid="loading">
      <div className="loading"></div>
    </S.Wrapper>
  )
}

export default Loading
