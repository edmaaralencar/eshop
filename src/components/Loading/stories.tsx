import { StoryObj, Meta } from '@storybook/react'
import Loading from '.'

export default {
  title: 'Loading',
  component: Loading
} as Meta

export const Default: StoryObj = {}
