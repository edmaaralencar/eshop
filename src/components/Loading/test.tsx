import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test-helper'

import Loading from '.'

describe('<Loading />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Loading />)

    expect(screen.getByTestId('loading')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
