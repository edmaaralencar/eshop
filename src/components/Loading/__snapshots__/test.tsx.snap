// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the heading 1`] = `
.c0 .loading {
  width: 10rem;
  height: 10rem;
  border: 5px solid #F4F4FF;
  border-radius: 50%;
  border-top-color: #8787FF;
  -webkit-animation: spinner 2s linear infinite;
  animation: spinner 2s linear infinite;
  margin: 0 auto;
}

<div
  class="c0"
  data-testid="loading"
>
  <div
    class="loading"
  />
</div>
`;
